lib/bolt/pal.rb

```diff
@@ -137,7 +137,9 @@ def in_bolt_compiler
       # TODO: If we always call this inside a bolt_executor we can remove this here
       setup
       r = Puppet::Pal.in_tmp_environment('bolt', modulepath: @modulepath, facts: {}) do |pal|
-        Puppet.override(bolt_project: @project,
+        # Only load the project if it a) exists, b) has a name it can be loaded with
+        bolt_project = @project if @project&.name
```

Comment:
I think we need to do this same thing in catalog.rb:
```
lucy ~/githubs/bolt ((5697f54...)) $ be bolt plan run bolt::foo -t localhost --verbose
No project name is specified in bolt-project.yaml. Project-level content will not be available.
Starting: plan bolt::foo
Starting: install puppet and gather facts on localhost
Finished: install puppet and gather facts with 0 failures in 2.29 sec
Starting: apply catalog on localhost
localhost:         Invalid module name ''; module names must match either:
        An installed module name (ex. modulename) matching the expression /^[a-z][a-z0-9_]*$/ -or-
        A namespaced module name (ex. author-modulename) matching the expression /^[a-zA-Z0-9]+[-][a-z][a-z0-9_]*$/
Started on localhost...
Finished on localhost:
  Notice: /Stage[main]/Main/Notify[foo]/message: defined 'message' as 'foo'
  changed: 1, failed: 0, unchanged: 0 skipped: 0, noop: 0
Finished: apply catalog with 0 failures in 3.89 sec
Finished: plan bolt::foo in 6.18 sec
Plan completed successfully with no result
```

```
bolt_project = Struct.new(:name, :path).new(project['name'], project['path']) unless project['name'].nil?
```

Reply:
Updated. I added the logic to the applicator to not include the project in the compilation 
 context if it doesn't have name.